src/node/rpc/frontend.h

```diff
@@ -870,7 +884,8 @@ namespace ccf
       update_history();
 
 #ifndef PBFT
-      bool is_primary = (consensus == nullptr) || consensus->is_primary();
+      bool is_primary = (consensus == nullptr) || consensus->is_primary() ||
+        ctx.is_create_request;
```

Comment:
This should go once we only call `process()` from `nodestate.h`

Reply:
discussed offline